tasks/Task.cpp

```diff
@@ -78,25 +78,34 @@ void Task::processIO()
     }
 
     /**
-     * Write received state to output port
-     * Only send control command after receiving a valid frame of an expected type coming from the generator
-     * If received frame is valid but not of an expected type or comes from another source, just ignore it and
+     * Generator starts running when it receives the start command. After that, keep sending the keep_alive command to the generator
+     * until receive stop control command on the input port.
+     * Only send control command after receiving a valid frame coming from the generator.
+     * If received frame is valid but comes from another source or has a different target, just ignore it and
      * don't send control command
      */
-    if (frame.targetID == variable_speed::TARGET_ADDRESS && frame.sourceID == variable_speed::SOURCE_ADDRESS) {
+    if (frame.targetID == variable_speed::PANELS_ADDRESS && frame.sourceID == variable_speed::DDC_CONTROLLER_ADDRESS) {
         if (frame.command == variable_speed::PACKET_GENERATOR_STATE_AND_MODEL){
             _generator_state.write(m_driver->parseGeneratorStateAndModel(frame.payload, now).first);
-
-            while (_control_cmd.read(m_control_cmd) == RTT::NewData) {
-                m_driver->sendControlCommand(m_control_cmd);
-            }
         }
         else if (frame.command == variable_speed::PACKET_RUN_TIME_STATE) {
             _run_time_state.write(m_driver->parseRunTimeState(frame.payload, now));
-
-            while (_control_cmd.read(m_control_cmd) == RTT::NewData) {
-                m_driver->sendControlCommand(m_control_cmd);
+        }
+        if (m_running) {
+            sentCommand = CONTROL_CMD_KEEP_ALIVE;
+        }
+        while (_control_cmd.read(m_control_cmd) == RTT::NewData) {
```

Comment:
You don't need both m_control_cmd and m_running. Actually, I think you should just keep m_control_cmd (renamed to m_last_command).

We read in attributes when the read value needs to be reused, or in critical codepaths to avoid memory allocations. But the control command has no allocation so you could have defined it as a local variable. 

Now, moving this logic to a separate method and removing m_running would dramatically reduce the number of branches (a good metric for "simplify"). In particular, it avoids the sentCommand temporary, which is really hard to follow:

In `updateHook`:

~~~
m_last_cmd = processStartStopCommand();
~~~

The new method:

~~~
ControlCommand Task::processStartStopCommand() {
    ControlCommand cmd;
    if (_control_cmd.read(cmd) == RTT::NoData) {
        return m_last_cmd;
    }

    if (cmd != m_last_cmd) {
        m_driver->sendControlCommand(cmd);
        return cmd
    }

    if (m_last_cmd == CONTROL_CMD_START) {
        m_driver->sendControlCommand(CONTROL_CMD_KEEPALIVE);
    }
    return m_last_cmd;
}
~~~